lightkurve/correctors/designmatrix.py

```diff
@@ -224,56 +252,101 @@ def append_constant(self, prior_mu=0, prior_sigma=np.inf):
-                          "Consider reducing the dimensionality by calling the "
-                          "`pca()` method.".format(self.rank, self.shape[1]),
-                          LightkurveWarning)
+        if rank:
+            if self.rank < (0.5*self.shape[1]):
+                warnings.warn("The design matrix has low rank ({}) compared to the "
+                              "number of columns ({}), which suggests that the "
+                              "matrix contains duplicate or correlated columns. "
+                              "This may prevent the regression from succeeding. "
+                              "Consider reducing the dimensionality by calling the "
+                              "`pca()` method.".format(self.rank, self.shape[1]),
+                              LightkurveWarning)
+        if self.prior_mu is not None:
+            if len(self.prior_mu) != self.shape[1]:
+                raise ValueError('`prior_mu` must have shape {}'
+                                 ''.format(self.shape[1]))
+        if self.prior_sigma is not None:
+            if len(self.prior_sigma) != self.shape[1]:
+                raise ValueError('`prior_sigma` must have shape {}'
+                                 ''.format(self.shape[1]))
+            if np.any(np.asarray(self.prior_sigma) <= 0):
+                raise ValueError('`prior_sigma` values cannot be smaller than '
+                                 'or equal to zero')
+
+    def validate(self, rank=True):
+        """Raises a `LightkurveWarning` if the matrix has a low rank, or priors that are the wrong shape."""
+        self._validate()
 
     @property
     def rank(self):
         """Matrix rank computed using `numpy.linalg.matrix_rank`."""
         return np.linalg.matrix_rank(self.values)
 
-    @property
-    def columns(self):
-        """List of column names."""
-        return list(self.df.columns)
-
     @property
     def shape(self):
         """Tuple specifying the shape of the matrix as (n_rows, n_columns)."""
-        return self.df.shape
+        return self.X.shape
 
     @property
     def values(self):
         """2D numpy array containing the matrix values."""
         return self.df.values
 
     def __getitem__(self, key):
-        return self.df[key]
+        return self.df[key].values
 
     def __repr__(self):
         return '{} DesignMatrix {}'.format(self.name, self.shape)
 
+    def to_sparse(self):
+        """ Convert a dense DesignMatrix object to a SparseDesignMatrix
+
+        The values of this design matrix will be converted to a
+        `scipy.sparse.csr_matrix`, which stores the values in a
+        lower memory matrix. This is not recommended for dense matrices.
+        """
+        return SparseDesignMatrix(csr_matrix(self.values), name=self.name,
+                            columns=self.columns, prior_mu=self.prior_mu,
+                             prior_sigma=self.prior_sigma)
```

Comment:
Let's align the indent with the opening delimiter here :point_up:, and perhaps only have one parameter on each line.